src/pysonar_scanner/api.py

```diff
@@ -102,24 +102,24 @@ def from_dict(dict: dict) -> "JRE":
             raise MissingKeyException(f"Missing key in dictionary {dict}") from e
 
 
-def get_base_urls(config_dict: dict[str, any]) -> BaseUrls:
-    @dataclass(frozen=True)
-    class ApiConfiguration:
-        sonar_host_url: str
-        sonar_scanner_sonarcloud_url: str
-        sonar_scanner_api_base_url: str
-        sonar_region: str
-
-    def to_api_configuration(config_dict: dict[str, any]) -> "ApiConfiguration":
-        return ApiConfiguration(
-            sonar_host_url=config_dict.get("sonar.host.url", ""),
-            sonar_scanner_sonarcloud_url=config_dict.get("sonar.scanner.sonarcloudUrl", ""),
-            sonar_scanner_api_base_url=config_dict.get("sonar.scanner.apiBaseUrl", ""),
-            sonar_region=config_dict.get("sonar.region", ""),
-        )
+ApiConfiguration = TypedDict(
+    "ApiConfiguration",
+    {"sonar.host.url": str, "sonar.scanner.sonarcloudUrl": str, "sonar.scanner.apiBaseUrl": str, "sonar.region": str},
+)
+
 
-    def is_sq_cloud_url(sonar_host_url: str) -> bool:
-        sq_cloud_url = api_config.sonar_scanner_sonarcloud_url.strip() or "https://sonarcloud.io"
+def to_api_configuration(config_dict: dict[str, any]) -> "ApiConfiguration":
```

Comment:
With a `TypedDict`, the return annotation doesn't need to be a string literal anymore (I was surprised to discover this was a thing, by the way...)